test/unit/spec/OktaSignIn_spec.js

```diff
@@ -250,6 +253,25 @@ Expect.describe('OktaSignIn initialization', function() {
         }
       });
     });
+    ['ready', 'afterError', 'afterRender'].forEach(event => {
+      it(`traps third party errors (for ${event} event) in callbacks`, function() {
+        const mockError = new Error('mockerror');
+        signIn.on(event, function() {
+          throw mockError;
+        });
+        signIn.trigger(event);
+        expect(Logger.error).toHaveBeenCalledWith(`[okta-signin-widget] "${event}" event handler error:`, mockError);
+      });
+    });
+    it('does not trap errors non-registered events', () => {
+      try {
+        signIn.on('not-widget-event', function() {
+          throw new Error('mockerror');
+        });
+        signIn.trigger('not-widget-event');
+        expect(Logger.error).not.toHaveBeenCalled();
+      } catch (err) {} // eslint-disable-line no-empty
```

Comment:
having a catch block may trap and ignore unexpected errors.  Try using `bind` to set up the function call and then use expect(fn).toThrow(...) to assert the specific mock error was thrown